src/components/ChatApp/Settings/Settings.react.js

```diff
@@ -108,826 +85,57 @@ class Settings extends Component {
-    });
-  };
-
-  // save new TTS settings
-  handleNewTextToSpeech = settings => {
-    this.setState({
-      speechRate: settings.speechRate,
-      speechPitch: settings.speechPitch,
-      ttsLanguage: settings.ttsLanguage,
-    });
-  };
-
-  // Handle toggle between default server and custom server
-  handleServeChange = event => {
-    let state = this.state;
-    let serverUrl;
-    if (event.target.value === 'customServer') {
-      state.checked = !state.checked;
-      defaults = UserPreferencesStore.getPreferences();
-      state.serverUrl = defaults.StandardServer;
-      state.serverFieldError = false;
-    } else if (event.target.name === 'serverUrl') {
-      serverUrl = event.target.value;
-      //eslint-disable-next-line
-      let validServerUrl = /(http|ftp|https):\/\/[\w-]+(\.[\w-]+)+([\w.,@?^=%&amp;:~+#-]*[\w@?^=%&amp;~+#-])?/i.test(
-        serverUrl,
-      );
-      state.serverUrl = serverUrl;
-      state.serverFieldError = !(serverUrl && validServerUrl);
-    }
-    this.setState(state);
-
-    if (this.state.serverFieldError) {
-      this.customServerMessage = 'Enter a valid URL';
-    } else {
-      this.customServerMessage = '';
-    }
-
-    if (this.state.serverFieldError && this.state.checked) {
-      this.setState({ validForm: false });
-    } else {
-      this.setState({ validForm: true });
-    }
-  };
-
-  // Close settings and redirect to landing page
-  onRequestClose = () => {
-    this.props.history.push('/');
-    window.location.reload();
-  };
-
-  // Open Remove Device Confirmation dialog
-  handleRemoveConfirmation = i => {
-    let data = this.state.obj;
-    let devicename = data[i].devicename;
-    this.setState({
-      showRemoveConfirmation: true,
-      showOptions: false,
-      removeDevice: i,
-      removeDeviceName: devicename,
-    });
-  };
-
-  // Show Top Bar drop down menu
-  showOptions = event => {
-    this.setState({
-      showOptions: true,
-      anchorEl: event.currentTarget,
-    });
-  };
-
-  handlePrefLang = event => {
-    this.setState({
-      PrefLanguage: event.target.value,
-    });
-  };
-
-  handleTimeZone = value => {
-    this.setState({
-      TimeZone: value,
-    });
-  };
-
-  // Close Top Bar drop down menu
-  closeOptions = () => {
-    this.setState({
-      showOptions: false,
-    });
-  };
-
-  componentWillUnmount() {
-    MessageStore.removeChangeListener(this._onChange);
-    UserPreferencesStore.removeChangeListener(this._onChangeSettings);
-  }
-
-  // Populate language list
-  _onChange = () => {
-    this.setState({
-      voiceList: voiceListChange,
-    });
-  };
-
-  // eslint-disable-next-line
   componentDidMount() {
-    document.body.className = 'white-body';
-    if (!this.state.dataFetched && cookies.get('loggedIn')) {
-      this.apiCall();
+    const { accessToken, actions } = this.props;
+    if (accessToken) {
+      actions.getUserSettings().then(() => {
+        this.setState({ loading: false, theme: this.props.theme });
```

Comment:
Does this.props.theme return the updated theme? Should we take the payload from the server API response in then block and then update it?

Reply:
@akshatnitd Yes, API returns theme. It is necessary to update the theme, I could I get it from the response or from props(store), similar stuff, right?

Reply:
Yes, it is similar. But I was checking, if you get the updated props in the then block or not.

Reply:
Tried it, it's behaving weirdly, loader keeps loading. Don't know why @akshatnitd